monkestation/code/modules/ghost_players/area_changes.dm

```diff
@@ -117,15 +117,12 @@
 /area/Entered(atom/movable/thing)
 	. = ..()
 
-	if(istype(thing, /mob/living/carbon/human/ghost))
-		// If this is a ghost, run the teleport checks
-		teleport_ghost_mob_if_needed(thing)
-	else
-		// Else, loop through this thing's contents...
-		for(var/atom/movable/atom_inside in thing.get_all_contents())
-			if(istype(atom_inside, /mob/living/carbon/human/ghost))
-				// ...and for any ghosts, run the checks
-				teleport_ghost_mob_if_needed(atom_inside)
+	// For the atom that just entered this area, as well as each of its contents...
+	for(var/atom/movable/atom_inside in thing.get_all_contents())
```

Comment:
Could do an as anything here

Reply:
Not sure if it needs it? We're only checking for atoms that are of type `/mob/living/carbon/human/ghost`, and teleporting those.

Reply:
I'm not sure why `as anything` would be needed here.

Reply:
`as anything` is useful here because it'll skip the implicit `/atom/movable` typecheck. Turfs and areas can't be inside a `contents` list, so we can safely assume that everything inside is an `/atom/movable` and so can safely skip the typecheck with `as anything`. I'm pushing for it here because `/area/Entered()` is a fairly hot proc, making every optimization worth it.

Reply:
Gotcha, thank you for explaining. Will make that change then.